test/val/val_id_test.cpp

```diff
@@ -3724,6 +3724,45 @@ OpFunctionEnd
           "ID 7 defined in block 6 does not dominate its use in block 9"));
 }
 
+TEST_F(ValidateIdWithMessage, SpecIdTargetNotSpecializationConstant) {
+  string spirv = kGLSL450MemoryModel + R"(
+OpDecorate %3 SpecId 200
+%1 = OpTypeVoid
+%2 = OpTypeFunction %1
+%int = OpTypeInt 32 0
+%3 = OpConstant %int 3
+%main = OpFunction %1 None %2
+%4 = OpLabel
+OpReturn
+OpFunctionEnd
+  )";
+  CompileSuccessfully(spirv.c_str());
+  EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateInstructions());
+  EXPECT_THAT(
+      getDiagnosticString(),
+      HasSubstr("OpDecorate SpectId decoration target <id> '1' is not a "
+                "scalar specialization constant."));
+}
```

Comment:
Suggest adding two cases, where the target is:
- an OpSpecConstantOp instruction
- an OpSpecConstnatComposite instruction

That's to prove that if an instruction name starts with "OpSpecConstant*" then it doesn't imply that it can be the target of a SpecId decoration.